src/Domain/FlaggingManager.php

```diff
@@ -19,34 +33,106 @@ public function checkModuleTranslations(array $moduleData, int $languageCount):
                 $flagData['moduleNeededTranslations'][] = $moduleTranslation['language']['name'];
             }
         }
-        if ((count($flagData['moduleNeededTranslations']) - $languageCount) <= -self::MIN_NEEDED_TRANSLATIONS) {
+        if (!in_array('English', $flagData['moduleNeededTranslations'], true) && (count($flagData['moduleNeededTranslations']) - $this->languageCount) <= -self::MIN_NEEDED_TRANSLATIONS) {
             $flagData['moduleStatus'] = true;
         }
 
+        $flagData['chapters'] = [];
+        foreach ($moduleData['chapters'] as $chapter) {
```

Comment:
Can we maybe split all these foreaches in their own private functions?
And then have 1 public function that calls them all.